### PR TITLE
Multi-arch release workflow modifications

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

#122  Modifies the create-release workflow to look for updates in multi-arch images.
- Polls for USNs/package updates in all supported architectures, in the off chance they are published separately
- Changes release asset structure to accomodate for multi-arch outputs. **See https://github.com/paketo-buildpacks/jammy-tiny-stack/releases/tag/untagged-985ef4e01877fc9c31b8 assets**
- Updates receipts.sh script to only output receipts as `<architecture>-<build/run>-receipt.cyclonedx.json`

## Use Cases

Make non-AMD64 architecture first class in workflows

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -26,55 +26,78 @@ env:
   RUN_RECEIPT_FILENAME: "run-receipt.cyclonedx.json"
   PATCHED_USNS_FILENAME: "patched-usns.json"
 jobs:
+  get_architectures:
+    name: Get Architectures
+    runs-on: ubuntu-22.04
```
Comment (robdimsdale) [inline .github/workflows/create-release.yml:31]: FYI `ubuntu-22.04` is now the same as `ubuntu-latest`. Just good to know.

File: .github/workflows/create-release.yml [hunk 2]
```diff
@@ -26,55 +26,78 @@ env:
...
+    outputs:
+      architectures: ${{ steps.lookup.outputs.platforms }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
```
Comment (robdimsdale) [inline .github/workflows/create-release.yml:36]: No rush, but we should bump to `actions/checkout@v4` as `v3` is deprecated.

File: .github/workflows/create-release.yml [hunk 3]
```diff
@@ -26,55 +26,78 @@ env:
...
+    - name: Lookup Supported Architectures
+      id: lookup
+      run: |
```
Comment (robdimsdale) [inline .github/workflows/create-release.yml:39]: Can we add the usual boiler plate here please?

```
#! /usr/bin/env bash

set -euo pipefail
shopt -s inherit_errexit
```

I can forsee a failure to download `yj` being confusing to debug :)

File: .github/workflows/create-release.yml [hunk 4]
```diff
@@ -172,20 +200,28 @@ jobs:
...
+                            --run-image "${GITHUB_WORKSPACE}/build/run.oci" \
+                            --build-receipt ${{ env.BUILD_RECEIPT_FILENAME }} \
+                            --run-receipt ${{ env.RUN_RECEIPT_FILENAME }}
+
+    - name: Upload ${{ matrix.arch }} Build receipt(s)
       uses: actions/upload-artifact@v3
       with:
         name: current-build-receipt
-        path: ${{ steps.receipts.outputs.build_receipt }}
+        path: "*${{ env.BUILD_RECEIPT_FILENAME }}"
```
Comment (robdimsdale) [inline .github/workflows/create-release.yml:215]: I see how the `*` works, but why do we need it? It looks like above we just write the file to: `--build-receipt ${{ env.BUILD_RECEIPT_FILENAME }}`

File: .github/workflows/create-release.yml [hunk 5]
```diff
@@ -283,24 +327,6 @@ jobs:
       env:
         BUILD_REMOVED: ${{ steps.build_diff.outputs.removed }}
         RUN_REMOVED: ${{ steps.run_diff.outputs.removed }}
-
-  run_if_packages_removed_with_force:
```
Comment (robdimsdale) [inline .github/workflows/create-release.yml:287]: I don't see where this has been moved to (or what it has been replaced with).

Do we still support the ability to force a release even if packages have been removed?

**Review by robdimsdale — COMMENTED** [review]
This looks great. I have a couple of questions/clarifications, and a couple of non-blocking observations. Mostly I want to make sure we continue to support all the permutations of forcing a release that we had before.

File: .github/workflows/create-release.yml (hunk 4 above)
Comment (sophiewigmore) [reply .github/workflows/create-release.yml:215]: The output of the Generate Package Receipts step file will be `<ARCH>-${{ env.BUILD_RECEIPT_FILENAME }}`, so in this case we will have `amd64-${{ BUILD_RECEIPT }}` and `arm64-${{ BUILD_RECEIPT }}`, so we need to upload both, which we can do using the `*`.

Comment (sophiewigmore) [reply .github/workflows/create-release.yml:215]: the `--build-receipt ${{ env.BUILD_RECEIPT_FILENAME }}` is just the file base name, but for all the files we generate we have an arch-specific version (receipts, release notes, usn file)

File: .github/workflows/create-release.yml (hunk 5 above)
Comment (sophiewigmore) [reply .github/workflows/create-release.yml:287]: As far as I could tell, this  particular job was ONLY checking if a package was removed with force, and printing a statement if it was.The logic to make sure we can force a release happens starting line 306 `- name: Fail If Packages Removed`, and I added  the print statement there if a package is removed with force! 

TLDR, this wasn't a super helpful code block and the core functionality you're asking about is still supported

File: .github/workflows/create-release.yml (hunk 3 above)
Comment (sophiewigmore) [reply .github/workflows/create-release.yml:39]: addressed in https://github.com/paketo-buildpacks/jammy-tiny-stack/compare/10213882af6cff681575c20de56649627a552a97..18bd7f895d73412ff2735dfc388f1084dcf9cd33

File: .github/workflows/create-release.yml (hunk 2 above)
Comment (sophiewigmore) [reply .github/workflows/create-release.yml]: addressed in https://github.com/paketo-buildpacks/jammy-tiny-stack/compare/10213882af6cff681575c20de56649627a552a97..18bd7f895d73412ff2735dfc388f1084dcf9cd33

File: .github/workflows/create-release.yml (hunk 4 above)
Comment (robdimsdale) [reply .github/workflows/create-release.yml:220]: ah ok. makes sense - thanks 👍

File: .github/workflows/create-release.yml (hunk 5 above)
Comment (robdimsdale) [reply .github/workflows/create-release.yml:287]: Gotcha. I see it now. Thanks for being patient with me 😄

**Review by robdimsdale — APPROVED** [review]
